conans/test/unittests/client/tools/cppstd_required_test.py

```diff
@@ -0,0 +1,183 @@
+                                 "compiler.version": version,
+                                 "compiler.cppstd": cppstd})
+        if libcxx:
+            settings.values["compiler.libcxx"] = libcxx
+        conanfile = MockConanfile(settings)
+        return conanfile
+
+    @parameterized.expand(["98", "gnu98", "11", "gnu11", "14", "gnu14", "17", "gnu17"])
+    def test_check_min_cppstd_from_settings(self, cppstd):
+        """ check_min_cppstd must accept cppstd less/equal than cppstd in settings
+        """
+        check_min_cppstd(self.conanfile, cppstd, False)
+
+    @parameterized.expand(["98", "gnu98", "11", "gnu11", "14", "gnu14"])
+    def test_check_min_cppstd_from_outdated_settings(self, cppstd):
+        """ check_min_cppstd must raise when cppstd is greater when supported on settings
+        """
+        self.conanfile.settings.values["compiler.cppstd"] = cppstd
+        with self.assertRaises(ConanInvalidConfiguration) as raises:
+            check_min_cppstd(self.conanfile, "17", False)
+        self.assertEqual("Current cppstd ({}) is lower than required c++ standard " 
+                         "(17).".format(cppstd), str(raises.exception))
+
+    @parameterized.expand(["98", "gnu98", "11", "gnu11", "14", "gnu14", "17", "gnu17"])
+    def test_check_min_cppstd_from_settings_with_extension(self, cppstd):
+        """ current cppstd in settings must has GNU extension when extensions is enabled
+        """
+        with mock.patch("platform.system", mock.MagicMock(return_value="Linux")):
+            with mock.patch.object(OSInfo, '_get_linux_distro_info'):
+                self.conanfile.settings.values["compiler.cppstd"] = "gnu17"
+                check_min_cppstd(self.conanfile, cppstd, True)
+
+                self.conanfile.settings.values["compiler.cppstd"] = "17"
+                with self.assertRaises(ConanInvalidConfiguration) as raises:
+                    check_min_cppstd(self.conanfile, cppstd, True)
+                self.assertEqual("Current cppstd (17) does not have GNU extensions, which is "
+                                 "required on Linux platform.", str(raises.exception))
+
+    @parameterized.expand(["98", "gnu98", "11", "gnu11", "14", "gnu14", "17", "gnu17"])
+    def test_check_min_cppstd_from_settings_with_extension_windows(self, cppstd):
+        """ GNU extensions has no effect on Windows for check_min_cppstd
+        """
+        with mock.patch("platform.system", mock.MagicMock(return_value="Windows")):
+            self.conanfile.settings.values["compiler.cppstd"] = "gnu17"
+            check_min_cppstd(self.conanfile, cppstd, True)
+
+            self.conanfile.settings.values["compiler.cppstd"] = "17"
+            check_min_cppstd(self.conanfile, cppstd, True)
+
+    def test_check_min_cppstd_unsupported_standard(self):
+        """ check_min_cppstd must raise when the compiler does not support a standard
+        """
+        del self.conanfile.settings.values["compiler.cppstd"]
```

Comment:
This should raise without removing the `compiler.cppstd` setting, right?

Reply:
Yes, but it will result as same as `test_check_min_cppstd_from_outdated_settings`

First, `check_min_cppstd` will check if the current conanfile.settings.compiler.cppstd < required cppstd. So, if it's true, it will raise an error. 

The test `test_check_min_cppstd_unsupported_standard` is using `42` which is higher.